cmd/juju/caas/add_test.go

```diff
@@ -15,22 +15,30 @@ import (
 	"github.com/juju/loggo"
 	jujutesting "github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/set"
 	gc "gopkg.in/check.v1"
 	"gopkg.in/juju/names.v2"
+	"gopkg.in/yaml.v2"
 
+	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/caas/kubernetes/clientconfig"
 	"github.com/juju/juju/cloud"
+	jujucloud "github.com/juju/juju/cloud"
 	"github.com/juju/juju/cmd/juju/caas"
+	jujucmdcloud "github.com/juju/juju/cmd/juju/cloud"
 	"github.com/juju/juju/jujuclient"
 )
 
 type addCAASSuite struct {
 	jujutesting.IsolationSuite
-	dir                 string
-	fakeCloudAPI        *fakeAddCloudAPI
-	store               *fakeCloudMetadataStore
-	fileCredentialStore *fakeCredentialStore
-	fakeK8SConfigFunc   clientconfig.ClientConfigFunc
+	dir                       string
+	fakeCloudAPI              *fakeAddCloudAPI
+	fakeK8sBrokerRegionLister *fakeK8sBrokerRegionLister
+	store                     *fakeCloudMetadataStore
+	fileCredentialStore       *fakeCredentialStore
+	fakeK8SConfigFunc         *clientconfig.ClientConfigFunc
+	currentClusterRegionSet   *set.Strings
+	isCloudRegionRequired     *bool
```

Comment:
This doesn't need to be a pointer.
Just set it to true when needed

Reply:
`isCloudRegionRequired` is passed into `fakeAddCloudAPI` in `SetUpTest`. This pointer allows to  reset value in actual test later after SetUpTest run, i think.

Reply:
ah, u r right. no need to have it in suite even.